src/plone/volto/tests/test_html2slate.py

```diff
@@ -0,0 +1,422 @@
+                        {"text": ""},
+                    ],
+                    "type": "li",
+                }
+            ],
+        )
+
+    def test_convert_slate_output_markup(self):
+        """test_convert_slate_output_markup."""
+        text = read_data("5.html")
+        res = text_to_slate(text)
+
+        self.assertEqual(
+            res,
+            read_json("5.json"),
+        )
+
+    def test_html2slate_list(self):
+        """test_slate_list."""
+        text = read_data("6.html")
+        res = text_to_slate(text)
+
+        self.assertEqual(
+            res,
+            read_json("6.json"),
+        )
+
+    def test_link_with_space(self):
+        html = "<li><a href='/'>      Black stork</a></li>"
+        res = text_to_slate(html)
+        self.assertEqual(
+            res,
+            [
+                {
+                    "children": [
+                        {"text": ""},
+                        {
+                            "children": [{"text": "Black stork"}],
+                            "data": {"url": "/"},
+                            "type": "link",
+                        },
+                        {"text": ""},
+                    ],
+                    "type": "li",
+                }
+            ],
+        )
+
+    def test_slate_data(self):
+        """test_slate_list."""
+        text = read_data("7.html")
+        res = text_to_slate(text)
+        self.assertEqual(
+            res,
+            read_json("7.json"),
+        )
+
+    def test_wrapped_slate_data(self):
+        """test_wrapped_slate_data."""
+        text = read_data("8.html")
+        res = text_to_slate(text)
+        self.assertEqual(
+            res,
+            read_json("8.json"),
+        )
+
+    def test_paragraph_sample(self):
+        """Encountered on first test with Volto"""
+        text = "<p>START <strong><em>partA</em></strong> <del>partB</del> END</p>"
+        res = text_to_slate(text)
+        self.assertEqual(
+            res,
+            [
+                {
+                    "children": [
+                        {"text": "START "},
+                        {
+                            "children": [
+                                {"text": ""},
+                                {"children": [{"text": "partA"}], "type": "em"},
+                                {"text": ""},
+                            ],
+                            "type": "strong",
+                        },
+                        {"text": " "},
+                        {"children": [{"text": "partB"}], "type": "del"},
+                        {"text": " END"},
+                    ],
+                    "type": "p",
+                }
+            ],
+        )
+
+    def test_ol_sample(self):
+        """Encountered while testing"""
+        text = read_data("9.html")
+        res = text_to_slate(text)
+        self.assertEqual(
+            res,
+            read_json("9.json"),
+        )
```

Comment:
It might make sense to use https://pypi.org/project/parameterized/ for these tests that are basically the same as each other except for the test data.

Reply:
@davisagli Good pointer, I knew about the parametrized tests for pytest, but not for the unittest. I actually briefly looked for a solution while developing the tests, but I felt that it would make the development harder, as I needed to work one test at a time, debug, etc. This would be harder with parametric tests. I think they're not that many anyway (though I hope we add more, if we find bugs).

Reply:
@tiberiuichim Fair point.